Sources/KlaviyoSwiftExtension/KlaviyoExtension.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  KlaviyoExtension.swift
+//  
+//
+//  Created by Ajay Subramanya on 6/23/23.
+//
+
+import Foundation
+import UIKit
+
+public struct KlaviyoExtensionSDK {
+    /// Default initializer for the Klaviyo SDK.
+    public init() {}
+    
+    /// Call this method when you receive a rich push notification in the notification service extension.
+    /// This method should be called from within `didReceive(_:withContentHandler:)` method of `UNNotificationServiceExtension`.
+    /// This method mainly does two things - downloads the media attached in the payload and then attaches it to the push notification.
+    ///
+    /// NOTE that there is no guarantees that the content handler will be called within the iOS stipulated time in the case where the download of the rich media takes long. iOS will automatically
+    /// present the notification as received from APNS when this happens and the image will not be attached to the notification.
+    ///
+    /// - Parameters:
+    ///   - request: the request received in the delegate `didReceive(_:withContentHandler:)`
+    ///   - bestAttemptContent: this is also received in `didReceive(_:withContentHandler:)` and is the best attempt at mutating the APNS payload before attaching it to the push notification
+    ///   - contentHandler: this is also received in `didReceive(_:withContentHandler:)` and is the closure that needs to be called before the time iOS provides for us to mutate the content. This closure will be called with the `bestAttemptContent` once the image is downloaded and attached.
+    public func handleNotificationServiceDidReceivedRequest(
+         request: UNNotificationRequest,
+         bestAttemptContent: UNMutableNotificationContent,
+         contentHandler: @escaping (UNNotificationContent) -> Void
+    ) {
+        // 1a. get the rich media url from the push notification payload
+        guard let imageURLString = bestAttemptContent.userInfo["rich-media"] as? String else {
+            contentHandler(bestAttemptContent)
+            return
+        }
+
+        // 1b.falling back to .png in case the media type isn't sent from the server.
+        let imageTypeString = bestAttemptContent.userInfo["rich-media-type"] as? String ?? "png"
```

Comment:
Default to png? Maybe should be a parameter with a default so they can override it?

Reply:
Good call. Done!